conversions/binary_to_decimal.py

```diff
@@ -37,6 +37,44 @@ def bin_to_decimal(bin_string: str) -> int:
     return -decimal_number if is_negative else decimal_number
 
 
+def bcd_to_gray(bcd):
+    """
+    Convert a Binary-Coded Decimal (BCD) number to Gray code.
+
+    Args:
+    bcd (int): The BCD number (0-9).
+
+    Returns:
+    str: The Gray code representation of the BCD number.
+
+    Raises:
+    ValueError: If the input is not a valid BCD number (0-9).
+    """
+
+    # Check if the input is a valid BCD (0-9)
+    if bcd < 0 or bcd > 9:
+        raise ValueError("Input must be a BCD number between 0 and 9.")
+
+    # Convert BCD to binary
+    binary = format(bcd, "04b")  # 4-bit binary representation
```

Comment:
We can simplify the conversion by using bit wise operators like this:

` gray = bcd ^ (bcd >> 1)  # XOR with right-shifted value
   return format(gray, '04b')`